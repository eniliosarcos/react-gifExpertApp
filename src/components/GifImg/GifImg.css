.card p {
  padding: 5px;
}
.hide{
  display: none;
}

img{
  margin-bottom: 10px;
  margin-right: 10px;
  max-height: 170px;
  border-radius: 5px;
}
