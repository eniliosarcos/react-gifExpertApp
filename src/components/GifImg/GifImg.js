import PropTypes from 'prop-types'
import { useEffect } from 'react';
import './GifImg.css'

const GifImg = ({ title, url }) => {

    const handleAnimation = (e) =>{
        if(e.target.complete){
            e.target.classList.remove('hide');
            e.target.classList.add('animate__animated', 'animate__fadeIn');
        }
    }

    return (
            <img className='hide' src={url} alt={title} onLoad={handleAnimation}></img>
    )
}

GifImg.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifImg
