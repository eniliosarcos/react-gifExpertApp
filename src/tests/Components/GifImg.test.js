import { fireEvent } from "@testing-library/react";
import { shallow } from "enzyme"
import GifImg from "../../components/GifImg/GifImg";


describe('Pruebas en Componente GifImg', () => {

    const title = "esto es un titulo";
    const url = "https://prueba.com";

    const wrapper = shallow(<GifImg title={title} url={url}></GifImg>);

    test('Debe de mostrar el componente correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        // console.log(img.props().src);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe de tener la imagen con clase hide',() =>{
        const img = wrapper.find('img');
        // console.log(img.props().src);
        expect(img.prop('className')).toBe('hide');
    });   
})
