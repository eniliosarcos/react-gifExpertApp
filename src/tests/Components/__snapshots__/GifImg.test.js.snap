// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Componente GifImg Debe de mostrar el componente correctamente 1`] = `
<img
  alt="esto es un titulo"
  className="hide"
  onLoad={[Function]}
  src="https://prueba.com"
/>
`;
